pkg/storage/replica.go

```diff
@@ -3055,21 +3064,59 @@ func (r *Replica) refreshProposalsLocked(refreshAtDelta int, reason refreshRaftR
 					fmt.Sprintf("unknown status for command without MaxLeaseIndex "+
 						"at refreshProposalsLocked time (refresh reason: %s)", reason)))})
 		} else if p.command.MaxLeaseIndex <= r.mu.state.LeaseAppliedIndex {
-			// The command's designated lease index range was filled up, so send it
-			// back to the proposer for a retry.
+			// The command's designated lease index slot was filled up. We got to
+			// LeaseAppliedIndex and p is still pending in r.mu.proposals; generally
+			// this means that proposal p didn't commit, and it will be sent back to
+			// the proposer for a retry - the request needs to be re-evaluated and the
+			// command re-proposed with a new MaxLeaseIndex.
+			//
+			// Except in the case when we're refreshing because of
```

Comment:
Awkward to start with "Except". Try a slight rephrase. Maybe "An exception is the case"